### DISCUSSION TITLE
How to use a GitHub repository as a webpage? （GitHub Pages）

### DISCUSSION THREAD

**Discussion by djl777** [community/community#70153] [Other Feature Feedback, Questions, & Ideas]
## Problem

I want to use a specific **GitHub Repository** to deploy a static webpage (as I've already used xxx.github.io). I use [docsify](https://docsify.js.org/#/) to deploy it. This is my related **repo**(https://github.com/djl777/note), and this is my **website**(https://djl777.github.io/note/#/) generated.

When I upload my folder, I find some trouble: my avg and font are not changed, the relative code is in https://github.com/djl777/note/blob/master/index.html: 
```html
 <link rel="icon" href="/static/image/avator.jpg">

<!-- font -->
  <link rel="stylesheet" href="/static/css/font.css">
```
So I change them to:
```html
 <link rel="icon" href="/note/static/image/avator.jpg">

<!-- font -->
  <link rel="stylesheet" href="/note/static/css/font.css">
```
and it does works, but does not work in my **local port** becauses of the ‘note’, so this is not a good idea.

I want to know why the root directory is xxx.github.io instead of xxx.github.io/note, and how to change it.

## supplement(another problem)
Today I find another problem in https://github.com/djl777/note/blob/master/README.md. 
```markdown
![test1](/static/image/avator.jpg)

![test2](/note/static/image/avator.jpg)
```
You could see my [homepage](https://djl777.github.io/note/#/) that the **test1(/static/image/avator.jpg) is ok but not test2(/note/static/image/avator.jpg)**, this disproves my previous conjecture about absolute paths, and I'm very confused as to why `index.html` behaves differently than `README.md`. Is this not a github page issue but something to do with docify?

Besides, my setting is:
![image](https://github.com/community/community/assets/73270782/be4cfd40-4171-4628-80d4-025ed85dcee6)

**Comment by sunny0625** [comment]
Ahh, Can you tell me the tech stack of your repo that you want to deploy on GitHub Pages?
Also, upload some screenshot to better understand your issue

**Reply by djl777** [reply]
I am very sorry . I have described the problem in more detail and I would appreciate if you could take a look at it for me.

**Comment by seyLu** [comment]
#### Hi @djl777,

Add a base url at the head of your html file:

```html
<head>
  ...
  <base href="/<repo>/">
</head>
```

<br>

This will resolve relative links to, for example:

```html
<head>
  ...
  <base href="/base_url/">
</head>
<body>
  ...
  <a href="some_link">
</body>
```

to

```bash
"/<base_url>/<some_link>"
```

<br>

You can read more at:
- [MDN docs: base url](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)

**Reply by djl777** [reply]
Thanks you very much. This does work in web but not in local port.
I have described the problem in more detail and I would appreciate if you could take a look at it for me.

**Comment by djl777** [comment]
Close
